java/runtime/src/main/java/io/ray/runtime/runner/RunManager.java

```diff
@@ -34,382 +24,99 @@
-        LOGGER.debug("Terminating process {}.", name);
-        p.destroy();
-      } else {
-        LOGGER.debug("Terminating process {} forcibly.", name);
-        p.destroyForcibly();
-      }
-      try {
-        p.waitFor(KILL_PROCESS_WAIT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
-      } catch (InterruptedException e) {
-        LOGGER.warn("Got InterruptedException while waiting for process {}" +
-            " to be terminated.", name);
-      }
-      numAttempts++;
+    Matcher matcher = pattern.matcher(output);
+    if (matcher.find()) {
+      String redisAddress = matcher.group(1);
+      rayConfig.setRedisAddress(redisAddress);
+    } else {
+      throw new RuntimeException("Redis address is not found. output: " + output);
     }
-    LOGGER.debug("Process {} is now terminated.", name);
+    LOGGER.info("Ray runtime started @ {}.", rayConfig.nodeIp);
   }
 
   /**
-   * Get processes by name. For test purposes only.
+   * Stop ray.
    */
-  public List<Process> getProcesses(String name) {
-    return processes.stream().filter(pair -> pair.getLeft().equals(name)).map(Pair::getRight)
-        .collect(Collectors.toList());
-  }
+  public static void stopRay() {
+    List<String> command = new ArrayList<>();
+    command.add("ray");
+    command.add("stop");
+    command.add("--force");
 
-  private void createTempDirs() {
     try {
-      FileUtils.forceMkdir(new File(rayConfig.logDir));
-      FileUtils.forceMkdir(new File(rayConfig.rayletSocketName).getParentFile());
-      FileUtils.forceMkdir(new File(rayConfig.objectStoreSocketName).getParentFile());
-
-      // Remove session_latest first, and then create a new symbolic link for session_latest.
-      final String parentOfSessionDir = new File(rayConfig.sessionDir).getParent();
-      final File sessionLatest = new File(
-          String.format("%s/%s", parentOfSessionDir, SESSION_LATEST));
-      if (sessionLatest.exists()) {
-        sessionLatest.delete();
-      }
-      Files.createSymbolicLink(
-          Paths.get(sessionLatest.getAbsolutePath()),
-          Paths.get(rayConfig.sessionDir));
-    } catch (IOException e) {
-      LOGGER.error("Couldn't create temp directories.", e);
-      throw new RuntimeException(e);
+      runCommand(command);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to stop ray.", e);
     }
   }
 
-  /**
-   * @return Log files for stdout and stderr.
-   */
-  private Pair<File, File> getLogFiles(String logDir, String processName) {
-    int suffixIndex = 0;
-    while (true) {
-      String suffix = suffixIndex == 0 ? "" : "." + suffixIndex;
-      File stdout = new File(String.format("%s/%s%s.out", logDir, suffix, processName));
-      File stderr = new File(String.format("%s/%s%s.err", logDir, suffix, processName));
-      if (!stdout.exists() && !stderr.exists()) {
-        return ImmutablePair.of(stdout, stderr);
-      }
-      suffixIndex += 1;
+  public static void fillConfigForDriver(RayConfig rayConfig) {
```

Comment:
this name is a bit ambiguous. something like `getAddressInfoAndFillConfig` would be better. also should document that this depends on executing an internal Python API.